src/qz/common/TrayManager.java

```diff
@@ -92,36 +95,33 @@ public TrayManager(boolean isHeadless) {
         shortcutCreator = DeployUtilities.getSystemShortcutCreator();
         shortcutCreator.setShortcutName(Constants.ABOUT_TITLE);
 
-        SystemUtilities.setSystemLookAndFeel();
+        // TODO: DON'T FORGET TO ADDRESS THIS. It causes a gtk2 vs 3 error on elementaryOS
+        // This at least mitigates the problem, still need to fix
+        if (!dorkbox.util.OS.isLinux() || dorkbox.systemTray.jna.linux.Gtk.isGtk2) {
+            SystemUtilities.setSystemLookAndFeel();
+        }
+
+        // Constructor iterates over all images denoted by IconCache.getTypes() and caches them
         iconCache = new IconCache();
 
         if (!headless) {
-            Image blank = new ImageIcon(new byte[1]).getImage();
-            if (SystemUtilities.isWindows()) {
-                tray = new JXTrayIcon(blank);
-            } else if (SystemUtilities.isMac()) {
-                tray = new ClassicTrayIcon(blank);
-            } else {
-                tray = new ModernTrayIcon(blank);
-            }
-
-            // Iterates over all images denoted by IconCache.getTypes() and caches them
-            tray.setImage(iconCache.getImage(IconCache.Icon.DANGER_ICON, tray.getSize()));
-            tray.setToolTip(name);
-
             try {
-                SystemTray.getSystemTray().add(tray);
+                tray = SystemTray.getNative();
+                tray.setImage(iconCache.getImage(IconCache.Icon.DANGER_ICON));
+                // tray.setToolTip(name); Not supported by Dorkbox (yet)
+            }
+            catch(Exception e ) {
+                log.error("Could not initialize tray, forcing headless mode", e);
+                headless = true;
             }
-            catch(AWTException awt) {
-                log.error("Could not attach tray, forcing headless mode", awt);
+            if (!headless && (tray == null)) {
+                log.error("Unknown tray init error, most likely unsupported os, Forcing headless mode");
```

Comment:
If the os doesn't support systemtrays, systemtray.get*() is returned null. This should catch every error except for Gtk errors that happen outside of the JVM.